src/primitives/cs_helpers.rs

```diff
@@ -54,18 +54,18 @@ pub fn constraint_evaluation(
         domain_size,
     );
 
-    let _ = boojum_cuda::gates::evaluate_gates(
-        &gates,
-        &variable_columns_matrix,
-        &witness_columns_matrix,
-        &constant_columns_matrix,
-        challenge,
-        &mut quotient_matrix,
-        challenge_power_offset as u32,
-        get_stream(),
-    )?;
-
-    Ok(())
+    if_not_dry_run! {
+        boojum_cuda::gates::evaluate_gates(
+            &gates,
+            &variable_columns_matrix,
+            &witness_columns_matrix,
+            &constant_columns_matrix,
+            challenge,
+            &mut quotient_matrix,
+            challenge_power_offset as u32,
+            get_stream(),
+        ).map(|_| ())
```

Comment:
hmm evaluate_gates updates `challenge_power_offset` internally, which we need to consider if we ever call evaluate_gates in a loop. Currently we only call it twice, once for general and once for specialized gates, with a pre-calculated challenge_power_offset argument each time, so i think it's ok to ignore the internal update.

Reply:
correct, in this case we are only interested in the error if there is one and can ignore the returned Ok value